Mage.Sets/src/mage/cards/s/ShareTheSpoils.java

```diff
@@ -0,0 +1,316 @@
+    }
+
+    private ShareTheSpoilsPlayExiledCardEffect(final ShareTheSpoilsPlayExiledCardEffect effect) { super(effect); }
+
+    @Override
+    public boolean applies(UUID sourceId, Ability source, UUID affectedControllerId, Game game) {
+        Card card = game.getCard(sourceId);
+        if (card == null) { return false; }
+
+        // Have to play on your turn
+        if (!game.getActivePlayerId().equals(affectedControllerId)) { return false; }
+
+        // Not in exile
+        if (game.getState().getZone(card.getMainCard().getId()) != Zone.EXILED) { return false; }
+
+        // TODO: This is a workaround for #8706, remove when that's fixed.
+        int zoneChangeCounter = game.getState().getZoneChangeCounter(source.getSourceId());
+        // Not a card exiled with this Share the Spoils
+        ExileZone exileZone = game.getExile().getExileZone(CardUtil.getExileZoneId(game, source.getSourceId(), zoneChangeCounter));
+        if (exileZone == null) { return false; }
+        if (!exileZone.contains(sourceId)) { return false; }
+
+        ShareTheSpoilsWatcher watcher = game.getState().getWatcher(ShareTheSpoilsWatcher.class);
+
+        return watcher.hasNotUsedAbilityThisTurn(new MageObjectReference(source));
+    }
+
+    @Override
+    public ShareTheSpoilsPlayExiledCardEffect copy() { return new ShareTheSpoilsPlayExiledCardEffect(this); }
+
+    @Override
+    public boolean apply(Game game, Ability source) { return true; }
+}
+
+//-- Spend mana as any color --//
+class ShareTheSpoilsSpendAnyManaEffect extends AsThoughEffectImpl implements AsThoughManaEffect {
+
+    ShareTheSpoilsSpendAnyManaEffect() {
+        super(AsThoughEffectType.SPEND_OTHER_MANA, Duration.WhileOnBattlefield, Outcome.Benefit);
+    }
+
+    private ShareTheSpoilsSpendAnyManaEffect(final ShareTheSpoilsSpendAnyManaEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) { return true; }
+
+    @Override
+    public ShareTheSpoilsSpendAnyManaEffect copy() { return new ShareTheSpoilsSpendAnyManaEffect(this); }
+
+    @Override
+    public boolean applies(UUID sourceId, Ability source, UUID affectedControllerId, Game game) {
+        Card card = game.getCard(sourceId);
+        if (card == null) { return false; }
+
+        // TODO: This is a workaround for #8706, remove when that's fixed.
+        int zoneChangeCounter = game.getState().getZoneChangeCounter(source.getSourceId());
+        // Check Exile
+        ExileZone exileZone = game.getExile().getExileZone(CardUtil.getExileZoneId(game, source.getSourceId(), zoneChangeCounter));
+        if (exileZone == null) { return false; }
+        if (exileZone.contains(sourceId)) { return true; }
+
+        // Check Stack
+        UUID cardID;
+        Card mainCard = card.getMainCard();
+        if (mainCard instanceof ModalDoubleFacesCard) {
+            cardID = ((ModalDoubleFacesCard) mainCard).getLeftHalfCard().getId();
```

Comment:
Will be broken for card's right side. Must check both sides. I recommends to search `card instanceof ModalDoubleFacesCard` for code examples. Also there are another multi side cards like split or adventure (don't know is it important here).

![shot_220218_012937](https://user-images.githubusercontent.com/8344157/154573952-87b9dffe-42f4-49ad-a7de-7f7142877251.png)

Good practice:
- targeting main card in all zones except battlefield;
- targeting object (specific card part on battlefield);
- checking main or all parts in the effect (`CardUtil.getObjectParts` can be usefull);
- see comments in `SPEND_OTHER_MANA` for usage instructions (e.g. about ZCC checking).